lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -1073,6 +1073,55 @@ Speculation::Speculatability ExternElementwiseOp::getSpeculatability() {
   return Speculation::NotSpeculatable;
 }
 
+// -- GatherOp --
+LogicalResult GatherOp::verify() {
+  RankedTensorType indicesTy = getIndices().getType();
+  RankedTensorType srcTy = getSrc().getType();
+  RankedTensorType resTy = getResult().getType();
+
+  if (indicesTy.getShape() != resTy.getShape()) {
```

Comment:
Could it be checked with `SameOperandsAndResultShape`?

Reply:
The source tensor can have a different shape/encoding from everything else though.

Reply:
Fair enough